crates/uv-trampoline-builder/src/lib.rs

```diff
@@ -486,6 +536,66 @@ if __name__ == "__main__":
         format!("#!{executable}")
     }
 
+    /// Creates a self-signed certificate and returns its path.
+    fn create_temp_certificate(temp_dir: &tempfile::TempDir) -> Result<PathBuf> {
+        use p12::PFX;
+        use rcgen::{CertificateParams, KeyPair};
+
+        let signing_key = KeyPair::generate()?;
+        let mut cert_params = CertificateParams::new(vec!["UvTrampolineTest".to_string()])?;
+        cert_params.insert_extended_key_usage(rcgen::ExtendedKeyUsagePurpose::CodeSigning);
+        let cert = cert_params.self_signed(&signing_key)?;
+
+        // Create PKCS#12 archive
+        let pfx = PFX::new(
+            cert.der(),
+            &signing_key.serialize_der(),
+            None,
+            "",
+            "UvTrampolineTest",
+        )
+        .expect("Failed to create PFX archive");
+
+        // Create temp file
+        let temp_pfx = temp_dir.path().join("uv-trampoline-test.pfx");
+        fs_err::write(&temp_pfx, pfx.to_der())?;
+
+        println!(
+            "Wrote testing code-signing certificate in {}",
+            temp_pfx.display()
+        );
+        Ok(temp_pfx)
+    }
+
+    /// Signs the given binary using `PowerShell`'s `Set-AuthenticodeSignature` with a temporary certificate.
+    fn sign_authenticode(bin_path: impl AsRef<Path>) {
+        let temp_dir = tempfile::tempdir().expect("Failed to create temporary directory");
```

Comment:
```suggestion
        let temp_dir = tempfile::TempDir::new().expect("Failed to create temporary directory");
```

nit: consistency w/ other uv crates